### PR TITLE
linuxPackages.rtl8814au: move to a more maintained fork

### PR DESCRIPTION
**Pull request by moni-dz**
###### Motivation for this change
ZHF: #122042 

move to a community maintained fork of rtl8814au
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fortuneteller2k** [comment]
Result of `nixpkgs-review pr 123632` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.rtl8814au</li>
    <li>linuxPackages_latest-libre.rtl8814au</li>
  </ul>
</details>
<details>
  <summary>14 packages built:</summary>
  <ul>
    <li>linuxPackages.rtl8814au</li>
    <li>linuxPackages_4_14.rtl8814au</li>
    <li>linuxPackages_4_19.rtl8814au</li>
    <li>linuxPackages_4_4.rtl8814au</li>
    <li>linuxPackages_4_9.rtl8814au</li>
    <li>linuxPackages_5_11.rtl8814au</li>
    <li>linuxPackages_5_12.rtl8814au</li>
    <li>linuxPackages_5_4.rtl8814au</li>
    <li>linuxPackages_hardened.rtl8814au</li>
    <li>linuxPackages_latest_hardened.rtl8814au</li>
    <li>linuxPackages_lqx.rtl8814au</li>
    <li>linuxPackages_testing_bcachefs.rtl8814au</li>
    <li>linuxPackages_xanmod.rtl8814au</li>
    <li>linuxPackages_zen.rtl8814au</li>
  </ul>
</details>

File: pkgs/os-specific/linux/rtl8814au/default.nix [hunk 1]
```diff
@@ -31,9 +31,8 @@ stdenv.mkDerivation rec {
 
   meta = with lib; {
     description = "Realtek 8814AU USB WiFi driver";
-    homepage = "https://github.com/zebulon2/rtl8814au";
-    license = licenses.gpl2;
+    homepage = "https://github.com/morrownr/8814au";
+    license = licenses.gpl2Only;
     maintainers = [ maintainers.lassulus ];
-    platforms = [ "x86_64-linux" "i686-linux" ];
```
Comment (Mindavi) [inline pkgs/os-specific/linux/rtl8814au/default.nix:37]: Why is this removed?

Comment (fortuneteller2k) [reply pkgs/os-specific/linux/rtl8814au/default.nix:37]: The fork supports `aarch64`, so no need to isolate platforms to `x86_64` and `i686`.

**Comment by Lassulus** [comment]
Result of `nixpkgs-review pr 123632` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages-libre.rtl8814au</li>
    <li>linuxPackages_hardkernel_4_14.rtl8814au</li>
    <li>linuxPackages_hardkernel_latest.rtl8814au</li>
    <li>linuxPackages_latest-libre.rtl8814au</li>
  </ul>
</details>
<details>
  <summary>15 packages built:</summary>
  <ul>
    <li>linuxPackages.rtl8814au (linuxPackages_5_10.rtl8814au)</li>
    <li>linuxPackages_4_14.rtl8814au</li>
    <li>linuxPackages_4_19.rtl8814au</li>
    <li>linuxPackages_4_4.rtl8814au</li>
    <li>linuxPackages_4_9.rtl8814au</li>
    <li>linuxPackages_5_11.rtl8814au (linuxPackages_latest.rtl8814au)</li>
    <li>linuxPackages_5_12.rtl8814au</li>
    <li>linuxPackages_5_4.rtl8814au</li>
    <li>linuxPackages_hardened.rtl8814au</li>
    <li>linuxPackages_latest_hardened.rtl8814au</li>
    <li>linuxPackages_lqx.rtl8814au</li>
    <li>linuxPackages_testing_bcachefs.rtl8814au</li>
    <li>linuxPackages_xanmod.rtl8814au</li>
    <li>linuxPackages_zen.rtl8814au</li>
    <li>tests.trivial</li>
  </ul>
</details>

**Review by Lassulus — APPROVED** [review]

File: pkgs/os-specific/linux/rtl8814au/default.nix (hunk 1 above)
Comment (Lassulus) [reply pkgs/os-specific/linux/rtl8814au/default.nix:37]: well since this can't be compiled without the kernel anyway, not setting it should be fine?
